### ISSUE TITLE
"Reset directory facets" link

### ISSUE THREAD

**Issue by willguv** [localgovdrupal/localgov_directories#111]
This was discussed and designed in Croydon - see below

- "Refine your search" label above facets
- "Clear all filters" link to reset all facets

As a citizen
I need a quick way to clear the facets I've selected on a directory
So I can see all the results again

**Comment by clearspotmedia** [comment]
![Directory-Search-Advanced-1170](https://user-images.githubusercontent.com/2931552/121044141-322cd280-c7ad-11eb-8cea-def02d0e669b.jpg)

Hi,  this was the original design with full functionality. At Croydon we launched with a cut back MVP version. 

It has a clear all filters link, clear buttons for each facet group. And also a show/hide filters button at the bottom

**Comment by willguv** [comment]
@clearspotmedia This is great - thank you!

**Comment by joachim-n** [comment]
There's also no way currently to clear the text search.

I'd say there's a need for a big 'Reset' button next to 'Apply', but I don't know about the UX implications of that -- would it be clear that it applies to the facets as well, or would it be taken as being only for the text?

**Comment by clearspotmedia** [comment]
I agree with @joachim-n that clearing the text search to see all results is not intuitive.

1) After carrying out a search, when the user focuses on the search input field, the place holder value should disappear

2) At the moment the user would have to empty the search bar and hit apply to show all results.
For that reason I'm in agreement a 'Reset search' link could be bundled into the search box.

My instinct says use a small link rather than another big button, as its secondary to the main purpose of searching and hitting apply.

**Comment by andybroomfield** [comment]
> When a user enables the option to 'Show reset link', the link is is not displayed. I strongly suspect this is because the processFacetList function in the DirectoryExtraFieldDisplay class inadvertently removes it when processing facet groups.
Originally by @MattOz-CDS  in #285
